app/src/main/java/de/westnordost/streetcomplete/screens/main/map/LocationAwareMapFragment.kt

```diff
@@ -93,10 +105,22 @@ open class LocationAwareMapFragment : MapFragment() {
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        displayedLocation = savedInstanceState?.getParcelable(DISPLAYED_LOCATION)
-        val nullTerminatedTracks = savedInstanceState?.getParcelableArrayList<Location?>(TRACKS) as ArrayList<Location?>?
-        if (nullTerminatedTracks != null) {
-            tracks = nullTerminatedTracks.unflattenNullTerminated()
+        // Restore value of members from saved state
+        if (savedInstanceState != null) {
+            with(savedInstanceState) {
+                displayedLocation = getParcelable(DISPLAYED_LOCATION)
+                tracksRecording = getBoolean(TRACKS_IS_RECORDING)
+                // It seems that the last list element will be an empty one normally
+                // If it is an empty one we can just remove it so we can keep on recording
+                val nullTerminatedTracks =
+                    getParcelableArrayList<Location?>(TRACKS) as ArrayList<Location?>?
+                if (nullTerminatedTracks != null) {
+                    tracks = nullTerminatedTracks.unflattenNullTerminated()
+                    if (tracksRecording && tracks.last().isEmpty()) {
```

Comment:
I am not sure if this is since a `unflattenNullTerminated` is used to encode / decode, but if you encode 2 elements in an array and then decode it, it will have a size of 3 with the last one being empty. If recording, we want to append locations to the last track, thus we can't have this new array be appended....

Reply:
Better add a comment in the code instead of in this PR only:

```suggestion
                    // unflattenNullTerminated creates an empty list item at the end
                    if (tracksRecording && tracks.last().isEmpty()) {
```

And why is the condition `tracksRecording && …`? Shouldn't the last empty list item always be removed?

Reply:
It should always be removed, if recording, yes.